### PR TITLE
Added support for HAProxy's PROXY protocol v2

### PR DESCRIPTION
**Pull request by CM8DoubleCheck**

### CODE REVIEW

**Review by haveachin — CHANGES_REQUESTED** [review]
Left you a few comments

**Review by dada513 — APPROVED** [review]

File: mc/conn.go [hunk 1]
```diff
@@ -53,12 +57,34 @@ func Dial(addr string) (Conn, error) {
 }
 
 // DialTimeout acts like DialMC but takes a timeout.
-func DialTimeout(addr string, timeout time.Duration) (Conn, error) {
+func DialTimeout(addr string, proxyProtocol bool, timeout time.Duration) (Conn, error) {
```
[the remark this answers is not in the document]
Comment (CM8DoubleCheck) [reply mc/conn.go:60]: No specific reason, I'll move it to the end

**Comment by CM8DoubleCheck** [comment]
I realized one oversight, I needed to pass the incoming client connection address to DialTimeout() rather than getting net.DialTimeout()'s remoteAddr (which would just return the connecting server's connection address). That's fixed and I verified it's working.

**Review by haveachin — APPROVED** [review]

**Comment by haveachin** [comment]
Thanks for your contribution :)

**Pull request merged**